dpnp/backend/src/dpnp_utils.hpp

```diff
@@ -98,6 +99,39 @@ void get_xyz_by_id_inkernel(size_t global_id, const _DataType* offsets, size_t o
     return;
 }
 
+/**
+ * @ingroup BACKEND_UTILS
+ * @brief Calculate xyz id for given axis from linear index
+ *
+ * Calculates xyz id of the array with given shape.
+ * for example:
+ *   input_array_shape_offsets[20, 5, 1]
+ *   global_id == 5
+ *   axis == 1
+ *   xyz_id should be 1
+ *
+ * @param [in]  global_id     linear index of the element in multy-D array.
+ * @param [in]  offsets       array with input offsets.
+ * @param [in]  offsets_size  array size for @ref offsets parameter.
+ * @param [in]  axis          axis.
+ */
+template <typename _DataType>
+_DataType get_xyz_id_by_id_inkernel(size_t global_id, const _DataType* offsets, size_t offsets_size, size_t axis)
```

Comment:
name get_xyz_by_id_inkernel_inloop?

Reply:
1. Maybe it is better to just modify "get_xyz_by_id_inkernel" ? Having to functions with particular same functionality confuses.

Reply:
variant 2
use this function inside get_xyz_by_id_inkernel() to avoid code duplication

Reply:
I like the variant 2.